node/cn/state_accessor.go

```diff
@@ -115,10 +115,14 @@ func (cn *CN) stateAtBlock(block *types.Block, reexec uint64, base *state.StateD
 	)
 	for current.NumberU64() < origin {
 		// Print progress logs if long enough time elapsed
-		if time.Since(logged) > 8*time.Second && report {
+		if report && time.Since(logged) > 8*time.Second {
 			logger.Info("Regenerating historical state", "block", current.NumberU64()+1, "target", origin, "remaining", origin-block.NumberU64()-1, "elapsed", time.Since(start))
 			logged = time.Now()
 		}
+		// Quit the state regeneration if time limit exceeds
+		if cn.config.DisableUnsafeDebug && time.Since(start) > cn.config.StateRegenerationTimeLimit {
```

Comment:
I think the same concurrency limit should be considered for this function as well. Doesn't it?

Reply:
Right, `stateAtBlock` is called by `StateAtBlock`, and it is called by `traceBlock` which is controlled by concurrency limit. So I think it can be said that it is controlled by concurrency limit.